File: example/r2.py
import msgpack
import requests


message = {
    "Action": "SAVE",
    "Bucket": "webui",
    "Data": {
        "_id":"r2-test-01.png",
        "path": "/Users/harry/5.png",
        "mime": "image/png"
    }
}

m = msgpack.packb(message, use_bin_type=True)


sess = requests.session()
sess.keep_alive = False
url = "http://localhost:9090/r2/action"

r = sess.post(url, data=m, auth=('admin','123'), stream=True)
print(r.text)

File: example/r2.php
<?php
ini_set('mongo.native_long', 1);

require_once("send_request.php");

$r2_data = array();

$r2_data["Bucket"] = "webui";
?>

<h2>SAVE</h2>
<?php

$r2_data["Action"] = "SAVE";
$r2_data["Data"] = array(
	 "_id" => "test/r2-php-test-01.png",
	 "path" => "/Users/harry/5.png",
	 "mime" => "image/png",
);



$res_save = PostMsgPackURL($url_r2_action,$r2_data);

print_r($res_save["info"]["http_code"]);
echo "<br>";
print_r($res_save["output"]);

?>

<h2>DELETE</h2>
<?php

$r2_data["Action"] = "DELETE";
$r2_data["Data"] = array(
	 "_id" => "test/r2-php-test-01.png",
);



$res_delete = PostMsgPackURL($url_r2_action,$r2_data);

print_r($res_delete["info"]["http_code"]);
echo "<br>";
print_r($res_delete["output"]);

?>